sklearn/utils/_logistic_sigmoid.pyx

```diff
@@ -11,17 +11,21 @@ cimport numpy as np
 ctypedef np.float64_t DTYPE_t
 
 
-cdef DTYPE_t _inner_log_logistic_sigmoid(DTYPE_t x):
+cdef inline DTYPE_t _inner_log_logistic_sigmoid(const DTYPE_t x):
     """Log of the logistic sigmoid function log(1 / (1 + e ** -x))"""
     if x > 0:
-        return -log(1 + exp(-x))
+        return -log(1. + exp(-x))
     else:
-        return x - log(1 + exp(x))
+        return x - log(1. + exp(x))
 
 
-def _log_logistic_sigmoid(int n_samples, int n_features, 
-                           np.ndarray[DTYPE_t, ndim=2] X,
-                           np.ndarray[DTYPE_t, ndim=2] out):
+def _log_logistic_sigmoid(int n_samples, int n_features,
+                          DTYPE_t[:, :] X,
+                          DTYPE_t[:, :] out):
+    cdef:
+        unsigned int i
+        unsigned int j
+
```

Comment:
There's no need to have an unsigned int here. i will never be bigger than n_samples which is an int.
Besides, it causes a warning of signed vs unsigned comparison. And I'm wondering if it doesn't cause a casting at each step of the loop. Are timings identical with ints ?

Reply:
Ok, I set everything to unsigned.

There's no cast (at least cython doesn't generate any cast). I didn't observe any speed difference